utilities/license-header/helper-functions.sh

```diff
@@ -20,7 +20,49 @@
 ##
 ## ---------------------------------------------------------------------
 
-# check license header in all files
-for file in $(find . -type f); do 
-  ./utilities/license-header/check-license-header.sh $file
-done
+get_comment_symbol_of_file_type()
+{
+  if [[ -f $1 ]]; then
+    filepath=$1
+  else
+    echo "No file provided!"
+    exit 1
+  fi
+
+  # determine file name without full path
+  filename=$(basename $filepath)
+
+  # examine file type and determine comment symbol
+  if [[ $filename == CMakeLists.txt || $filename == *.cmake || $filename == *.sh ]]; then
+    comment="##"
+  elif [[ $filename == *.h || $filename == *.h.in || $filename == *.cc ]]; then
+    comment="\/\/"
+  elif [[ $fileext == *.rst ]]; then
+    comment=".."
+  else
+    comment=""
+  fi
+
+  echo $comment
+}
+
+generate_range_of_years()
+{
+  if [[ ! -z $1 && ! -z $2 ]]; then
+    yearadded=$1
+    yearlastmod=$2
+  else
+    echo "No year(s) provided!"
+    exit 1
+  fi
+
+  if [[ "$yearadded" == "$yearlastmod" ]]; then
+    years=$yearadded
+  else
+    years=$yearadded-$yearlastmod
+  fi
+
+  echo $years
+}
+
+exit 0
```

Comment:
Sorry for the (endless) failing workflows. Now I found out that this `exit 0` here is important when loading this file via `source utilities/license-header/helper-functions.sh`.